src/javaRosa.js

```diff
@@ -1523,8 +1407,80 @@ define([
 
             this.__callOld();
 
+            delete this.data.javaRosa.itextMap;
             Itext.on('change', function () { _this.onFormChange(); });
         },
+        populateControlMug: function(mug, controlElement) {
+            this.__callOld();
+
+            var Itext = this.data.javaRosa.Itext,
+                itextMap = this.data.javaRosa.itextMap;
+
+            function getITextID(value) {
+                try {
+                    var parsed = xpath.parse(value);
+                    if (parsed instanceof xpathmodels.XPathFuncExpr &&
+                        parsed.id === "jr:itext")
+                    {
+                        return parsed.args[0].value;
+                    }
+                } catch (err) {
+                    // this seems like a real error since the reference should presumably
+                    // have been valid xpath, but don't deal with it here
+                }
+                return "";
+            }
+
+            function getItextItem(id, property) {
+                var auto = !id || id === getDefaultItextId(mug, property);
+                if (id) {
+                    var item = itextMap[id];
+                    if (item && itextMap.hasOwnProperty(id)) {
+                        if (!auto) {
+                            item.autoId = false;
```

Comment:
All itext items have `autoId` turned on by default. This turns it off selectively for any ID that differs from the default. This used to be done later (during XML writing) leaving lots opportunity for the mug path to get out of sync with the itext ID.

From this point forward we depend on the `autoId` property of the itext rather than matching it ID against the default itext id for the mug/property.